pkgs/applications/version-management/stgit/default.nix

```diff
@@ -40,25 +56,28 @@ python3Packages.buildPythonApplication rec {
     substituteInPlace Documentation/texi.xsl \
       --replace http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd \
                 ${docbook_xml_dtd_45}/xml/dtd/docbook/docbookx.dtd
-
-    cat > stgit/_version.py <<EOF
-    __version__ = "${version}"
-    EOF
   '';
 
-  makeFlags = [
+  makeFlags = lib.strings.concatStringsSep " " [
     "prefix=${placeholder "out"}"
     "MAN_BASE_URL=${placeholder "out"}/share/man"
     "XMLTO_EXTRA=--skip-validation"
+    "PERL_PATH=${perl}/bin/perl"
   ];
 
-  buildFlags = [ "all" "doc" ];
+  buildPhase = ''
+    make all ${makeFlags}
+  '';
+
+  checkPhase = ''
+    make test ${makeFlags}
+  '';
```

Comment:
We don't need to manually define this, it should just work with makeFlags. Also runHook is missing

Reply:
I just tried, it doesn't. But it turns out, most of the flags are not necessary anymore, I'll clean them up and add the runHooks with the next release of stgit